### PR TITLE
properly handle parsing of miner init sector sizes

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by whyrusleeping** [comment]
still not sure how i ended up making a miner with 64GB sector size though. Something still seems off

**Comment by whyrusleeping** [comment]
Hrm.. something seems non-deterministic here:
```
why@sirius ~/c/lotus (fix/init-sector-size)> ./lotus-storage-miner init --help
NAME:
   lotus-storage-miner init - Initialize a lotus storage miner repo

USAGE:
   lotus-storage-miner init [command options] [arguments...]

OPTIONS:
   --actor value                specify the address of an already created miner actor
   --create-worker-key          create separate worker key (default: false)
   --worker value, -w value     worker key to use (overrides --create-worker-key)
   --owner value, -o value      owner key to use
   --sector-size value          specify sector size to use (default: "32GiB")
   --pre-sealed-sectors value   specify set of presealed sectors for starting as a genesis miner
   --pre-sealed-metadata value  specify the metadata file for the presealed sectors
   --nosync                     don't check full-node sync status (default: false)
   --symlink-imported-sectors   attempt to symlink to presealed sectors instead of copying them into place (default: false)
   --no-local-storage           don't use storageminer repo for sector storage (default: false)
   --gas-price value            set gas price for initialization messages in AttoFIL (default: "0")
   --help, -h                   show help (default: false)

why@sirius ~/c/lotus (fix/init-sector-size)> ./lotus-storage-miner init --help
NAME:
   lotus-storage-miner init - Initialize a lotus storage miner repo

USAGE:
   lotus-storage-miner init [command options] [arguments...]

OPTIONS:
   --actor value                specify the address of an already created miner actor
   --create-worker-key          create separate worker key (default: false)
   --worker value, -w value     worker key to use (overrides --create-worker-key)
   --owner value, -o value      owner key to use
   --sector-size value          specify sector size to use (default: "512MiB")
   --pre-sealed-sectors value   specify set of presealed sectors for starting as a genesis miner
   --pre-sealed-metadata value  specify the metadata file for the presealed sectors
   --nosync                     don't check full-node sync status (default: false)
   --symlink-imported-sectors   attempt to symlink to presealed sectors instead of copying them into place (default: false)
   --no-local-storage           don't use storageminer repo for sector storage (default: false)
   --gas-price value            set gas price for initialization messages in AttoFIL (default: "0")
   --help, -h                   show help (default: false)
```

**Comment by whyrusleeping** [comment]
found it. The slice sorter is comparing the same item with itself...

File: build/params_shared.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func DefaultSectorSize() abi.SectorSize {
 	}
 
 	sort.Slice(szs, func(i, j int) bool {
-		return szs[i] < szs[i]
+		return szs[i] < szs[j]
```
Comment (arajasek) [inline build/params_shared.go:28]: This looks like it might be more correct, yeah

**Pull request merged**